File: app/src/main/java/khorunzhyicom/alex/theposter/di/components/AppComponent.java
package khorunzhyicom.alex.theposter.di.components;

import javax.inject.Singleton;

import dagger.Component;
import khorunzhyicom.alex.theposter.di.components.injector.AppInjector;
import khorunzhyicom.alex.theposter.di.modules.ApiModule;
import khorunzhyicom.alex.theposter.di.modules.AppModule;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent extends AppInjector, ContextProvider {

    ApiComponent plus(ApiModule apiModule);
}

File: app/src/main/java/khorunzhyicom/alex/theposter/di/components/AppComponent.kt
package khorunzhyicom.alex.theposter.di.components

import dagger.Component
import khorunzhyicom.alex.theposter.di.components.injector.AppInjector
import khorunzhyicom.alex.theposter.di.modules.*
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent : AppInjector {

    fun create(): ServiceComponent
}
